### PR TITLE
use AddressAutocomplete on all address fields

### PR DESCRIPTION
**Pull request by tglaz**
**Client Profile - Add & Clean Up Address Suggestions**
https://betterangels.atlassian.net/browse/DEV-803

* Only Handles AC:
  * ALL address fields have suggestions provided by google map api as the user types
* The other AC about scrolling to be handled in separate PR

**Note**: I explored integrating [react-native-google-places-autocomplete](https://www.npmjs.com/package/react-native-google-places-autocomplete) but ran into some issues. 
Should be a good idea to revisit.

## Summary by Sourcery

Implement AddressAutocomplete for all address fields to provide Google Maps API suggestions, enhancing user experience and code maintainability.

New Features:
- Introduce AddressAutocomplete component to provide address suggestions using Google Maps API for all address fields.

Enhancements:
- Refactor address input fields to use the new AddressAutocomplete component, simplifying the code and improving user experience.

### CODE REVIEW

File: apps/betterangels/package.json [hunk 1]
```diff
@@ -79,7 +79,8 @@
     "expo-file-system": "*",
     "expo-sharing": "*",
     "@teovilla/react-native-web-maps": "*",
-    "react-native-paper": "*"
+    "react-native-paper": "*",
+    "@types/google.maps": "*"
```
Comment (vecchp) [inline apps/betterangels/package.json:83]: I think this is a dev dependency.

Comment (tglaz) [reply apps/betterangels/package.json:83]: So after i reinstall it as a devDep, `sync-deps` still puts it back under regular dependencies (in this file). 
Also, is there a process/command for uninstalling a package so it unsyncs it? Running sync-depds after removing it didn't do anything.

Comment (vecchp) [reply apps/betterangels/package.json:83]: let's take a look today after standup.  If it's automatically doing it then i guess we don't have a choice.

**Review by mikefeldberg — APPROVED** [review]
love this refactor. would like @Davit-BetterAngels to review as well, in case I missed anything

File: libs/expo/betterangels/src/lib/services/PlaceAutocomplete/getPlaceAutocomplete.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    query,
+    boundsCenter = LA_COUNTY_CENTER,
+    boundsRadiusMiles = 10,
+  } = props;
+
+  if (query.length < 3) {
+    return [];
+  }
+
+  // 1 degree of lat/long is approximately 69 miles
```
[the remark this answers is not in the document]
Comment (tglaz) [reply libs/expo/betterangels/src/lib/services/PlaceAutocomplete/getPlaceAutocomplete.ts:25]: yeah, i agree. for me this fits the exception where you use comments to explain the `why`

File: libs/expo/betterangels/src/lib/ui-components/AddressField/AddressAutocomplete.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import { Control, Controller, FieldValues, Path } from 'react-hook-form';
+import {
+  TPlaceResult,
+  TPlacesPrediction,
+  getPlaceAutocomplete,
+  getPlaceDetailsById,
+} from '../../services';
+import { AddressOption } from './AddressOption';
+
+const DEFAULT_DEBOUNCE_MS = 300;
```
Comment (tglaz) [inline libs/expo/betterangels/src/lib/ui-components/AddressField/AddressAutocomplete.tsx:14]: @vecchp -  I'd suggest changing `300ms` timeout to `100ms`
300 is annoying, and I'd even go 50ms until we actually notice a pricing issue.

If it becomes expensive then perhaps that's a good sign.
I think it's 2.83 USD per 1000 per [this pricing page](https://developers.google.com/maps/documentation/places/web-service/usage-and-billing), but **** google and their docs 🤷

File: apps/betterangels/package.json (hunk 1 above)
Comment (tglaz) [reply apps/betterangels/package.json]: done. i moved it manually

**Review by Davit-BetterAngels — APPROVED** [review]

**Pull request merged**